tstest/integration/testcontrol/testcontrol.go

```diff
@@ -67,6 +68,20 @@ func (s *Server) NumNodes() int {
 	return len(s.nodes)
 }
 
+// AddPingRequest adds the entry to the pingReqsToAdd map
+func (s *Server) AddPingRequest(nodeKeyDst tailcfg.NodeKey, pr *tailcfg.PingRequest) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	if s.pingReqsToAdd == nil {
+		s.pingReqsToAdd = map[tailcfg.NodeKey]*tailcfg.PingRequest{}
+	}
+	s.pingReqsToAdd[nodeKeyDst] = pr
+	// Now send the update to the channel
+	nodeID := s.Node(nodeKeyDst).ID
```

Comment:
I think you'll want to check whether Node returns nil, if so, don't add the ping to the map, and then return false (add a "bool" return value to AddPingRequest)